File: Swift/power-of-two.swift
class Solution {
    func isPowerOfTwo1(_ n: Int) -> Bool {
        var x = 1

        while x != n {
            if x < n {
                x <<= 1
            } else {
                return false
            }
        }

        return true
    }

    func isPowerOfTwo2(_ n: Int) -> Bool {
        return (n > 0) && (n & (n - 1) == 0)
    }
}

File: Python/power-of-two.py
class Solution:
    def isPowerOfTwo1(self, n):
        """
        :type n: int
        :rtype: bool
        """
        x = 1

        while x != n:
            if x < n:
                x <<= 1
            else:
                return False

        return True

    def isPowerOfTwo2(self, n):
        """
        :type n: int
        :rtype: bool
        """
        return (n > 0) and (n & (n - 1) == 0)
